junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -113,9 +138,11 @@ public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod
 	 */
 	class ReplaceUnderscores extends Standard {
 
+		static final DisplayNameGenerator INSTANCE = new ReplaceUnderscores();
+
 		@Override
 		public String generateDisplayNameForClass(Class<?> testClass) {
-			return replaceUnderscores(super.generateDisplayNameForClass(testClass));
+			return replaceUnderscores(testClass.getSimpleName());
```

Comment:
Nop this was a very interesting change indeed that I asked in glitter, 
when generating the report of a test with `replaceUnderscores` : 

	@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
	static class StackTestCase {

		Stack<Object> stack;

		@Test
		void new_stack() {
			new Stack<>();
		}
        [...]
     }
I got an output like this: 
<img width="501" alt="evidence" src="https://user-images.githubusercontent.com/17280833/71784424-10785c00-2fb9-11ea-9ce1-be5c442386f2.png">

I don't know if this is a desired output with the class parents with a $, but it seems odd to read that if the goal is making a result pretty to read. 

All of this is because`generateDisplayNameForClass` in Standard uses `class.getName()` in `generateDisplayNameForClass` and then strips the packages name with a `LastIndexOf('.')`, for inherence `super` also Underscores had the same output issue and then also Indicative Sentences because it to concatenate the results of the selected Generator.

I tried to make this change on Standard but to many test has this kind of result _ContainerClass$MyClassTest_ expected, so decided to let Standard  be and make this change.

Reply:
I see what you mean

@sormuras Was there a reason to reuse the `Standard` implementation in `ReplaceUnderscores`? Do you think this change would make sense?

Reply:
The idea was to keep the "relative path to the package" of a test class intact. Using `getSimpleName()` may result in many similar test names.

Reply:
All right I'll keep the result of the generators with the relative path of the package, and the proper changes on the test's.